output/extensions/registry.go

```diff
@@ -0,0 +1,64 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2021 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package extensions
```

Comment:
I'm not sure if this should be called "extensions" or be in a separate package. Like I mentioned in another comment, I don't think we should have a distinction between built-in and external outputs, since they all should implement the `output.Output` interface and are treated the same internally.

Also being in `output/extensions` makes it seem like these are extensions _for_ outputs, not additional outputs that are loaded externally.

Reply:
See comment above (https://github.com/loadimpact/k6/pull/1869#discussion_r582765131), I think there is value in distinguishing between built-in and extension modules, even if they all implement the same interface.

Regarding the name, I am not sure it's the best, but this doesn't seem all that confusing: 
```go
extensions.Register("outputName", OutputConstructor)
```

I guess I can move this to the parent module (`/output`) and rename it a bit, like this:
```go
output.RegisterExtension("outputName", OutputConstructor)
```

cc @MStoykov thoughts?

Reply:
I am finer with this being called `outputs.Register` and possibly having an `internal` packages as with the modules if we want to distinguish them. 

Although apparently @na-- is against us using the same method for both internal an external modules/outputs :man_shrugging:

Reply:
So you want me to change the module name from `output` to `outputs` and move `Register` there? I don't mind the second, but I'm against the rename to `outputs`

Regarding having a distinction between built-in and extension outputs, feel free to disagree in https://github.com/loadimpact/k6/pull/1869#discussion_r582765131

Reply:
`output` is fine ... just `outputs.Register` was more like English so this is what I wrote, but I agree that the package is better of being called `output`

Reply:
Is this better: https://github.com/loadimpact/k6/pull/1869/commits/5e41f5e8af0116ee3d5cbb048b0a1dde992e2a83 ?